File: c++_implemention/linear_search.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
#define doura ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);
#define sesh '\n';


int linear_search(int arr[], int n, int x) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == x) {
            return i;
        }
    }
    return -1;
}

int main()
{
    
    int n, x;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    cin >> x;

    int result = linear_search(arr, n, x);

    if (result != -1) {
        cout << "Element found at the index: " << result << sesh;
    }
    else {
        cout << "Element not found\n";
    }

    return 0;
}

File: js_implemention/linear_search.js

function linear_search(arr, n, x) {
    for (let i = 0; i < n; i++) {
        if (arr[i] == x) {
            return i;
        }
    }
    return -1;
}


function main() {
    let n = 5;
    let arr = [1, 2, 3, 4, 5];
    let x = 3;

    const result = linear_search(arr, n, x);
    
    if (result != -1) {
        console.log(`Element found at the index: ${result}`);
    }
    else {
        console.log(`Element not found!!`);
    }
}

main();
